src/gui/entry/EditEntryWidget.cpp

```diff
@@ -578,46 +585,95 @@ void EditEntryWidget::removeCurrentAttribute()
+
+    if (index.isValid()) {
+        QString key = m_attributesModel->keyByIndex(index);
+        if (showProtected) {
+            m_advancedUi->attributesEdit->setPlainText(tr("[PROTECTED] Press reveal to view or edit"));
+            m_advancedUi->attributesEdit->setEnabled(false);
+            m_advancedUi->revealAttributeButton->setEnabled(true);
+            m_advancedUi->protectAttributeButton->setChecked(true);
         }
         else {
-            m_advancedUi->attributesEdit->setPlainText("");
-            m_advancedUi->attributesEdit->setEnabled(false);
+            m_advancedUi->attributesEdit->setPlainText(m_entryAttributes->value(key));
+            m_advancedUi->attributesEdit->setEnabled(true);
+            m_advancedUi->revealAttributeButton->setEnabled(false);
+            m_advancedUi->protectAttributeButton->setChecked(false);
         }
+
+        // Don't allow editing in history view
+        m_advancedUi->protectAttributeButton->setEnabled(!m_history);
+        m_advancedUi->editAttributeButton->setEnabled(!m_history);
+        m_advancedUi->removeAttributeButton->setEnabled(!m_history);
     }
     else {
-        if (m_currentAttribute != newIndex) {
-            if (m_currentAttribute.isValid()) {
-                QString key = m_attributesModel->keyByIndex(m_currentAttribute);
-                m_entryAttributes->set(key, m_advancedUi->attributesEdit->toPlainText(),
-                                       m_entryAttributes->isProtected(key));
-            }
-
-            if (newIndex.isValid()) {
-                QString key = m_attributesModel->keyByIndex(newIndex);
-                m_advancedUi->attributesEdit->setPlainText(m_entryAttributes->value(key));
-                m_advancedUi->attributesEdit->setEnabled(true);
-            }
-            else {
-                m_advancedUi->attributesEdit->setPlainText("");
-                m_advancedUi->attributesEdit->setEnabled(false);
-            }
-
-            m_advancedUi->editAttributeButton->setEnabled(newIndex.isValid());
-            m_advancedUi->removeAttributeButton->setEnabled(newIndex.isValid());
-            m_currentAttribute = newIndex;
+        m_advancedUi->attributesEdit->setPlainText("");
+        m_advancedUi->attributesEdit->setEnabled(false);
+        m_advancedUi->revealAttributeButton->setEnabled(false);
+        m_advancedUi->protectAttributeButton->setChecked(false);
+        m_advancedUi->protectAttributeButton->setEnabled(false);
+        m_advancedUi->editAttributeButton->setEnabled(false);
+        m_advancedUi->removeAttributeButton->setEnabled(false);
+    }
+
+    m_advancedUi->protectAttributeButton->blockSignals(false);
+}
+
+void EditEntryWidget::protectCurrentAttribute(bool state)
+{
+    QModelIndex index = m_advancedUi->attributesView->currentIndex();
+    if (!m_history && index.isValid()) {
+        QString key = m_attributesModel->keyByIndex(index);
+        if (state)
```

Comment:
This should have braces, because with the comment lines, it looks more like two lines.